src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -115,8 +149,12 @@ internal static SourcePropertySymbol Create(SourceMemberContainerTypeSymbol cont
                 location,
                 diagnostics)
         {
-            if (IsAutoProperty)
+            if (IsAutoPropertyWithoutAccessorBinding)
```

Comment:
>IsAutoPropertyWithoutAccessorBinding [](http://example.com/codeflow?start=16&length=36)

```isAutoProperty```?

Reply:
@AlekseyTs I don't think these are equivalent. `isAutoProperty` is mutated in `SourcePropertySymbolBase` by:

```
isAutoProperty = isAutoProperty && !(containingType.IsInterface && !IsStatic) && !IsAbstract && !IsExtern && !isIndexer;
```

My change is supposed to keep the original behavior. Need to add corresponding tests, if there are not already.

Reply:
@AlekseyTs The following *existing* test will fail if `isAutoProperty` is used for the reason I mentioned:

https://github.com/dotnet/roslyn/blob/148f9e0ad71baad905b312e2463a07e35256fcec/src/Compilers/CSharp/Test/Symbol/Symbols/Source/PropertyTests.cs#L2930-L2961

The suggested change will cause the diagnostic to be reported on the property in the interface and abstract class.

Reply:
@AlekseyTs I made `isAutoProperty` a ref parameter to get rid of the `IsAutoPropertyWithoutAccessorBinding` property.

Reply:
Actually I think I don't need to do it that way. Checking for `_propertyFlags` is just sufficient. Will revert the `ref` parameter change.

Reply:
Hmm, doing it with `_propertyFlags` will require exposing the private `Flags` enum as well as the `_propertyFlags` field. I see each approach has its own advantages and disadvantages.

Will let you decide which approach is better.